qiskit_nature/second_q/operators/second_quantized_op.py

```diff
@@ -71,7 +71,22 @@ def is_hermitian(self, atol: Optional[float] = None) -> bool:
         Returns:
             True if the operator is hermitian up to numerical tolerance, False otherwise.
         """
+        return self.equiv(self.adjoint(), atol=atol)
+
+    def equiv(self, other: Any, atol: Optional[float] = None) -> bool:
```

Comment:
Since this class extends the `TolerancesMixin`, could we simply use this implementation for the `__eq__` method?
After all, that is what I would expect to happen for "tolerant" objects when comparing them...

(@woodsp-ibm: looping you in on this discussion)

Reply:
I think `__eq__` should be reserved for "true" equality (comparing coefficients using `==`). This behavior would be consistent with the way `__eq__` is typically defined (e.g. for built-in types and numpy arrays). If `equiv` is added then making `__eq__` redirect to it would be redundant and also preclude adding the "true" equality I just described.

Reply:
How do you suggest we implement "true" equality then? Comparing floats to an exact equality without some tolerance can be very flaky and simply checking whether two objects are actually identical instances is usually not what one is after when doing an `==` comparison.

To strengthen my argument I'd also like to point out, that this is exactly how `TolerancesMixin` is being used in Terra:
https://github.com/Qiskit/qiskit-terra/blob/316eaf5da1c3ff1c34d018c5aaca72e88a43483b/qiskit/quantum_info/states/statevector.py#L109-L112

https://github.com/Qiskit/qiskit-terra/blob/316eaf5da1c3ff1c34d018c5aaca72e88a43483b/qiskit/quantum_info/states/densitymatrix.py#L113-L116

Reply:
> How do you suggest we implement "true" equality then?

I think this should work:
```python
def __eq__(self, other):
    return self.terms == other.terms
```

> Comparing floats to an exact equality without some tolerance can be very flaky

I agree that when dealing with floats approximate equality is generally more useful than exact equality, but there are still occasions to use exact equality. Note that `a == b` means exact equality when `a` and `b` are `floats`.

> simply checking whether two objects are actually identical instances is usually not what one is after when doing an == comparison.

I disagree. For all built-in types as well as Numpy arrays, `==` means exact equality.

> To strengthen my argument I'd also like to point out, that this is exactly how TolerancesMixin is being used in Terra:

As I pointed out above, Terra is deviating here from the behavior of all built-in types as well as Numpy, so I would argue that this is surprising behavior which we should not emulate.

Reply:
Alright. I had another discussion on this with @woodsp-ibm offline.

We might add `__eq__` in the future, using true equality checking. However, we think that your new method here would be suitably named `isclose` rather than `equiv`. So please update the naming accordingly

Reply:
I'm fine with changing the name, but I just wanted you to be aware that I named it `equiv` to be consistent with `SparsePauliOp.equiv` which is the corresponding method on `SparsePauliOp`. Are you sure you want to break this consistency?

Reply:
Thanks for clarifying, I had overlooked that.
Let's stick with `equiv` then :+1: